src/Dapr.Client/ConfigurationSource.cs

```diff
@@ -0,0 +1,21 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Dapr.Client
+{
+    /// <summary>
+    /// Abstraction around a configuration source.
+    /// </summary>
+    public abstract class ConfigurationSource : IAsyncEnumerable<IEnumerable<ConfigurationItem>>
+    {
+        /// <summary>
+        /// The Id associated with this configuration source.
+        /// </summary>
+        public virtual string Id => string.Empty;
```

Comment:
What's this for?

Reply:
This should be abstract not virtual. I assume `string.Empty` is not a valid value for `Id`?

Reply:
Yeah I didn't realize that you could make a property abstract but not a member variable, made it abstract.